include/pybind11/pythoncapi_compat.h

```diff
@@ -0,0 +1,346 @@
+// Header file providing new functions of the Python C API to old Python
```

Comment:
Any reason for this not to be a submodule? Would make it much easier to update it for newer Python versions and such?

Reply:
We had a few submodules for a while and subdued in a library intended to be a submodule doesn’t go well. Mostly made gitlab users mad. ;) Everyone was happy to see them go. 

It’s harder to fork and customize in emergencies, too; changing the url can lead to issues.

I love submodules, but probably not for pybind11 itself. 

A single file is not bad, git subtree isn’t bad either.